app/src/main/java/com/example/taskyy/data/local/room_entity/UserEntity.kt

```diff
@@ -6,9 +6,8 @@ import androidx.room.PrimaryKey
 @Entity(tableName = "user_table")
 data class UserEntity(
     @PrimaryKey(autoGenerate = true)
-    val id: Int? = null,
-    val name: String,
-    val password: String,
-    val email: String
+    var id: Int? = null,
+    var name: String,
+    var email: String
```

Comment:
This is likely why you're not getting any response from the server. The class you send to the server needs to match exactly what the server expects. Same fields and same field names. When looking at the API docs, the /register endpoint accepts this format 
```
{
    "fullName": <String>
    "email": <String>
    "password": <String>
}
```
This doesn't match with your `UserEntity` which you're currently sending to the server. Check my next comment on how to fix that